### PR TITLE
remove deprecated skip_equivalent from pyproj

### PR DESCRIPTION
**Pull request by wang-boyu**
Fix https://github.com/projectmesa/mesa-geo/issues/39 due to https://github.com/pyproj4/pyproj/pull/824

### CODE REVIEW

File: mesa_geo/geospace.py [hunk 1]
```diff
@@ -96,7 +94,7 @@ def get_intersecting_agents(self, agent, other_agents=None):
         return intersecting_agents
 
     def get_neighbors_within_distance(
-        self, agent, distance, center=False, relation="intersects"
+            self, agent, distance, center=False, relation="intersects"
```
Comment (rht) [inline mesa_geo/geospace.py:97]: This is an unnecessary indentation.

Comment (wang-boyu) [reply mesa_geo/geospace.py]: Thank you for spotting this! It's probably due to the auto formating tool from my IDE.

**Comment by rht** [comment]
Merged as https://github.com/projectmesa/mesa-geo/commit/75d356e28e3134e888d7b58732d42fb7331a3746.
I was about to ask you to squash your commits, but figured it'd be faster if I squash it myself.

**Comment by rht** [comment]
Hmmm, while it is common for some FOSS projects to merge it the way I just did, https://github.com/projectmesa/mesa-geo/commit/75d356e28e3134e888d7b58732d42fb7331a3746 doesn't link to this PR. This way, generating the changelog items will be harder, where we usually link to the relevant PRs. I'm not going to do this in the future.

**Comment by wang-boyu** [comment]
> Hmmm, while it is common for some FOSS projects to merge it the way I just did, [75d356e](https://github.com/projectmesa/mesa-geo/commit/75d356e28e3134e888d7b58732d42fb7331a3746) doesn't link to this PR. This way, generating the changelog items will be harder, where we usually link to the relevant PRs. I'm not going to do this in the future.

This is interesting to know - all my previous projects (proprietary software) were using the merge method with `--no-ff`, but I'm fine with alternative ways.

In the future for `mesa` and `mesa-geo`, would you suggest to squash all commits before submitting a PR?

If yes then perhaps we can also standardise and mention it in the CONTRIBUTING.rst.

**Comment by rht** [comment]
Squashing all commits is problematic, see e.g.  https://github.com/JuliaDynamics/Agents.jl/commit/b1cd6a8671c593b1a33a54a054fd3cf779b11723, which is a giant commit where the intent of each individual change is hard to read. I favor atomic commits that each logically do different things. Which means, small changes with nonsensical commit message should be squashed into the closest commit that does the same thing. It is a bit too much to expect every contributor to do this fine-grained squashing, but then again, it is super rare for a contributor to create a large PR in the first place.

**Comment by rht** [comment]
Though you could argue that in https://github.com/JuliaDynamics/Agents.jl/commit/b1cd6a8671c593b1a33a54a054fd3cf779b11723, you can still see individual commits in the pull requests.

**Comment by wang-boyu** [comment]
Yeah I agree that squashing is problematic, guess this is why I seldom use it over the years.

On the other hand, if individual commits are preserved and the rebase method is used for pull requests, then the history of the main branch will be full of these small commits from different branches intertwined. This is basically why we (my previous projects) preferred the merge method with `--no-ff` for PRs. But again, I'm fine with any protocol so long as it's mutually agreed.

**Comment by rht** [comment]
I just remembered one reason keeping the commits separate instead of squashed for each PR is worthwhile: doing git-bisect to find a commit that causes a bug is much easier, vs narrowing it down to 1 giant commit.

**Comment by wang-boyu** [comment]
Yup!

**Pull request closed without merging**